client/client.go

```diff
@@ -963,3 +984,93 @@ func (r *NotaryRepository) DeleteTrustData() error {
 	}
 	return nil
 }
+
+// ListRootCerts returns current trusted root certificates.
+// (In particular, not old certificates which we have already
+// rotated from but are still using for signing to allow rollover,
+// even if they are still trusted locally.)
+func (r *NotaryRepository) ListRootCerts() ([]*x509.Certificate, error) {
+	_, err := r.Update(false)
+	if err != nil {
+		return nil, err
+	}
+
+	keyIDs := r.tufRepo.Root.Signed.Roles[data.CanonicalRootRole].KeyIDs
+	certs := make([]*x509.Certificate, 0, len(keyIDs))
+	for _, keyID := range keyIDs {
+		// r.tufRepo contains all data from root.json, which may include irrelevant
+		// certificates.  This is handled by certs.ValidateRoot, which adds
+		// only the real trusted certificates into the store.  So, this lookup by ID is not
+		// just a dumb way to not parse the certificate, the handling of ErrNoCertificatesFound
+		// ensures we only get the intersection of (certificates in current root.json) with
+		// (locally trusted certificates). The rotation code in certs.ValidateRoot ensures
+		// that we locally store a copy of all relevant certificates.
+		cert, err := r.CertStore.GetCertificateByCertID(keyID)
+		if err == nil {
+			certs = append(certs, cert)
+		} else if _, ok := err.(*trustmanager.ErrNoCertificatesFound); !ok {
+			return nil, err
+		}
+	}
+	return certs, nil
+}
+
+// RotateRootCert generates a new certificate to replace oldCert and adds
+// a changelist entry to update the root role with this certificate replacement
+// when r.Publish() is called.
+func (r *NotaryRepository) RotateRootCert(oldCert *x509.Certificate) error {
+	oldCertX509PublicKey := trustmanager.CertToKey(oldCert)
+	if oldCertX509PublicKey == nil {
+		return fmt.Errorf("invalid format for root key: %s", oldCert.PublicKeyAlgorithm)
+	}
+	rootKeyID, err := trustmanager.X509PublicKeyID(oldCertX509PublicKey)
```

Comment:
This returns the TUF ID rather than the canonical ID. If I remember right, `CryptoService.GetPrivateKey` requires the canonical ID.